### ISSUE TITLE
Swagger response validation not working correctly for 204 responses

### ISSUE THREAD

**Issue by dannylevenson** [BlueOakJS/blueoak-server#82]
In my swagger definition, I have the following responses defined for a PUT endpoint:
```
  responses:
    204:
      description: The password has been successfully reset.
    401:
      $ref: '../../../public/responses/Unauthorized.yaml'
    422:
      $ref: '../../../public/responses/InvalidRequest.yaml'
```

When my code issues a 204 response with no body, I see the following in the console:
```
error: -swagger - Response validation error: {
  "message": "No response schema defined for PUT /app-api/v2/customer-portal/users/sfico1/password with status code 204",
  "status": 522,
  "type": "ValidationError",
  "invalidResponse": {
    "method": "PUT",
    "path": "/app-api/v2/customer-portal/users/sfico1/password",
    "statusCode": 204,
    "body": null
  }
}
```

It appears that the Swagger response validation is not handling a 204 response correctly.

**Comment by ryshep111** [comment]
Looks like the code is requiring you to define a schema within the response object, even though the specification does not require it. Good catch Danny, will write a fix for this right away.

**Comment by ryshep111** [comment]
PR #83 submitted

**Comment by dannylevenson** [comment]
I am still seeing this issue in the ESB.  Here the exception:
```
error: errors - Other error:  TypeError: Cannot read property '$ref' of undefined
    at ValidatorContext.resolveRefs (/Users/danny/workspaces/SFIgit/sfi-esb/node_modules/tv4/tv4.js:399:12)
    at ValidatorContext.validateAll (/Users/danny/workspaces/SFIgit/sfi-esb/node_modules/tv4/tv4.js:536:16)
    at Object.api.validateMultiple (/Users/danny/workspaces/SFIgit/sfi-esb/node_modules/tv4/tv4.js:1597:12)
    at Object.exports.validateJSONType (/Users/danny/workspaces/SFIgit/sfi-esb/node_modules/blueoak-server/lib/swaggerUtil.js:23:22)
    at validateResponseModels (/Users/danny/workspaces/SFIgit/sfi-esb/node_modules/blueoak-server/handlers/_swagger.js:586:30)
    at ServerResponse.res.send (/Users/danny/workspaces/SFIgit/sfi-esb/node_modules/blueoak-server/handlers/_swagger.js:352:44)
    at /Users/danny/workspaces/SFIgit/sfi-esb/handlers/customer-portal.js:319:29
    at _fulfilled (/Users/danny/workspaces/SFIgit/sfi-esb/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/Users/danny/workspaces/SFIgit/sfi-esb/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/Users/danny/workspaces/SFIgit/sfi-esb/node_modules/q/q.js:796:13)
    at /Users/danny/workspaces/SFIgit/sfi-esb/node_modules/q/q.js:604:44
    at runSingle (/Users/danny/workspaces/SFIgit/sfi-esb/node_modules/q/q.js:137:13)
    at flush (/Users/danny/workspaces/SFIgit/sfi-esb/node_modules/q/q.js:125:13)
    at nextTickCallbackWith0Args (node.js:420:9)
    at process._tickDomainCallback (node.js:390:13)
```

Line 319 of the customer-portal handler is:
`res.status(204).send();`

@seanpk When I first installed, BOS in the ESB's node_modules was still 2.5.0.  I ran `npm install blueoak-server@2.6.0`, verified that BOS was actually updated.  I retried the scenario and still get this failure.

**Comment by ryshep111** [comment]
looking at the code, I see why it didn't work because it never aborts the validation in the case of an empty response body/no schema. What I don't understand is how the test didn't catch that.

**Comment by ryshep111** [comment]
PR #84 submitted

**Comment by seanpk** [comment]
should be fixed now, see v2.6.1
